apps/astarte_realm_management/lib/astarte_realm_management/queries.ex

```diff
@@ -209,41 +217,25 @@ defmodule Astarte.RealmManagement.Queries do
     end)
   end
 
-  def check_astarte_health(client, consistency) do
-    schema_statement = """
-      SELECT count(value)
-      FROM #{CQLUtils.realm_name_to_keyspace_name("astarte", Config.astarte_instance_id!())}.kv_store
-      WHERE group='astarte' AND key='schema_version'
-    """
-
-    # no-op, just to check if nodes respond
-    # no realm name can contain '_', '^'
-    realms_statement = """
-    SELECT *
-    FROM #{CQLUtils.realm_name_to_keyspace_name("astarte", Config.astarte_instance_id!())}.realms
-    WHERE realm_name='_invalid^name_'
-    """
+  def check_astarte_health(consistency) do
+    keyspace = Realm.keyspace_name("astarte")
 
     schema_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(schema_statement)
-      |> DatabaseQuery.consistency(consistency)
+      from KvStore,
+        where: [group: "astarte", key: "schema_version"],
+        limit: 1
 
+    # no-op, just to check if nodes respond
+    # no realm name can contain '_', '^'
     realms_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(realms_statement)
-      |> DatabaseQuery.consistency(consistency)
+      from RealmManagement.Astarte.Realm,
+        where: [realm_name: "_invalid^name_"],
+        limit: 1
 
-    with {:ok, result} <- DatabaseQuery.call(client, schema_query),
-         ["system.count(value)": _count] <- DatabaseResult.head(result),
-         {:ok, _result} <- DatabaseQuery.call(client, realms_query) do
+    with {:ok, _} <- Repo.fetch_one(schema_query, prefix: keyspace, consistency: consistency),
+         {:ok, _} <- Repo.fetch_one(realms_query, prefix: keyspace, consistency: consistency) do
```

Comment:
While I guess that the first query should always return a result, the `realms_query` as it is defined will never match any row and `Repo.fetch_one` will return `{:error, :not_found}`.